src/containers/clock.jsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+// import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import { Link } from 'react-router-dom';
+import Paper from '@material-ui/core/Paper';
+import Fab from '@material-ui/core/Fab';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import List from '@material-ui/core/List';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import Chip from '@material-ui/core/Chip';
+
+import TimelineIcon from '@material-ui/icons/Timeline';
+import { Typography } from '@material-ui/core';
+
+import WatchList from '../components/watch-list';
+import Stopwatch from '../components/time';
+import sortByTime from '../utils/sort';
+
+
+const styles = theme => ({
+  bebe: {
+    fontSize: 32,
+  },
+  root: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    minHeight: '100vh',
+    backgroundColor: '#333',
+    textAlign: 'center',
+    flexDirection: 'column',
+  },
+  paper: {
+    margin: theme.spacing(4),
+    padding: `${theme.spacing(3)}px ${theme.spacing(4)}px ${theme.spacing(4)}px`,
+    width: '100%',
+    maxWidth: theme.leftDrawerWidth,
+  },
+  contentLogo: {
+    width: '100%',
+    maxWidth: theme.leftDrawerWidth,
+  },
+  logo: {
+    width: '100%',
+    display: 'block',
+    margin: `0 auto ${theme.spacing(1)}px`,
+  },
+});
+
+const StopwatchControls = ({ onToggle, running}) =>
+  <Fab onClick={onToggle}>
+    {running ? 'STOP' : 'START'}
+  </Fab>
+
+const bestTime = (attempts) => {
+  if (attempts.length === 0){
+    return '00:00:00';
+  }
+  return Stopwatch(attempts.sort(sortByTime)[0])
+}
+
+const AverageTime = (attempts) => {
+  if (attempts.length === 0){
+    return '00:00:00';
+  }
+  const average = attempts.reduce((acc, current) => acc + current.time, 0) / attempts.length;
+  return Stopwatch({ time: average });
+}
+
+
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      time: 0,
+      running: false,
+      date: '',
+    };
+    this.toggle = this.toggle.bind(this);
+    this.tick = this.tick.bind(this);
+  }
+  toggle() {
+    if (!this.state.running) {
+      this.startTime = +new Date();
+      this.tick();
+    } else {
+      const dataLocal = JSON.parse(localStorage.getItem('attempts'));
```

Comment:
@MaiaRojas What happens if the user has denied access to `localStorage`?

Reply:
good catch :point_up: